mllib/src/main/scala/org/apache/spark/ml/tree/impl/DTStatsAggregator.scala

```diff
@@ -33,11 +34,13 @@ private[spark] class DTStatsAggregator(
 
   /**
    * [[ImpurityAggregator]] instance specifying the impurity type.
-   */
-  val impurityAggregator: ImpurityAggregator = metadata.impurity match {
+  */
+  private val impurityAggregator: ImpurityAggregator = metadata.impurity match {
+    // TODO(vlad17): this is a ridiculous coupling. Impurity should have a getAggregator() method
```

Comment:
It should, but it should not have a getAggregator() method which returns a class from spark.mllib.  getAggregator should be put inside Impurity after spark.ml has the necessary class---but that need should be documented here.

Can you please create a JIRA to fix this and modify your comment here to link to the new JIRA?  I'd also note in the JIRA that we should consider not having Aggregator classes in spark.ml at all since that functionality could live inside Impurity classes.